### DISCUSSION TITLE
Csla readonly collections' performance or .NET?

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3418] [Questions]
We are asked to replace an older MS Access application. The main form in the application is congested (and convoluted). We have around 20 combo boxes with four of the combo boxes have 4-5k items.
No matter what User Interface technology we use in .NET, the time it takes to load the form (and hence combo boxes ) is anywhere from 10 seconds to 20 seconds (I just tried Winforms).
We cache the read-only lists while the application loads so in the main form the collections' data is coming from the cache. We tried loading the combo boxes using async/await and even the older backgroundworker threads.
Nothing is making a difference : )

Here is the funny part, the older Access form load almost instantaneously lol.
What gives?

Kind regards

**Comment by rockfordlhotka** [comment]
I'd do some logging or other instrumentation so you can determine if the perf lag is in the server-side code where the list is loaded with data, or on the client side when it is data bound.

**Reply by rockfordlhotka** [reply]
If it is a UI binding thing, which it might be, then I'd look for a third party control that does lazy or windowed loading. I bet that's what Access does - it displays the control without loading all the data into the UI.

**Reply by Chicagoan2016** [reply]
You are right, it's not csla objects. I downloaded and ran the dottrace tool from JetBrains. 
I am looking at the winform UI and this form has a number of data bound controls (e.g combo box controls).
Even if I don't bind the combo box controls and remove every data access call (no csla object fetches) the form still takes three, four seconds to load.
My guess is, since the number of controls is large, they are taking time to 'paint'.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I started using Telerik controls and it helped. I also recommended UI changes in the future. 
This one (freakish) form touches almost every single table in the database : )
